system/jlib/jmisc.cpp

```diff
@@ -696,21 +696,31 @@ class AbortHandlerInfo : public CInterface
 public:
     ThreadId installer;
     AbortHandler handler;
+    AbortHandlerBase bhandler;
     IAbortHandler *ihandler;
     AbortHandlerInfo(AbortHandler _handler)
     {
         handler = _handler;
+        bhandler = NULL;
+        ihandler = NULL;
+        installer = GetCurrentThreadId();
+    }
+    AbortHandlerInfo(AbortHandlerBase _handler)
+    {
+        handler = NULL;
+        bhandler = _handler;
         ihandler = NULL;
         installer = GetCurrentThreadId();
     }
     AbortHandlerInfo(IAbortHandler *_ihandler)
     {
         handler = NULL;
+        bhandler = NULL;
         ihandler = _ihandler;
         installer = GetCurrentThreadId();
     }
 
-    bool handle()
+    bool handle(ahType aht_val)
     {
 #ifndef _WIN32
         if (installer == GetCurrentThreadId())
```

Comment:
I'm not sure I understand and agree with this statement.  I know its not related to this PR, but this suggests that the signal is handled only if it is received by the same thread that installed it, otherwise it is disposed.  If we want only the same thread to always handle it - shouldn't we set the mask of all other threads to block it ?  I think in modern Linux a signal is attempted to be delivered to the root/main/initial thread first and perhaps that is where it is installed and signals for us mean the end of the process soon anyway but still not sure if this is how it should be ?